yubikey-luks-enroll

```diff
@@ -44,27 +49,40 @@ if [ $CLEAR_SLOT -eq 1 ]; then
 	echo "Killing LUKS slot $SLOT"
 	cryptsetup luksKillSlot "$DISK" "$SLOT"
 fi
+
 echo "Adding yubikey to initrd"
 P1=$(/lib/cryptsetup/askpass "Please insert a yubikey and enter a new password. This is the password that will only work while your yubikey is installed in your computer.")
+	if [ "$DBG" = "1" ]; then echo "Password: $P1"; fi
+
 P2=$(/lib/cryptsetup/askpass "Please enter the yubikey password again:")
+	if [ "$DBG" = "1" ]; then echo "Password: $P2"; fi
+
 if [ "$P1" != "$P2" ]; then
 	echo "Passwords do not match"
 	exit 1
+fi
 
 if [ "$HASH" = "1" ]; then
-	P1=$(printf %s "$P1" | sha256sum | awk '{print $1}') 
+	P1=$(printf %s "$P1" | sha256sum | awk '{print $1}')
+		if [ "$DBG" = "1" ]; then echo "Password hash: $P1"; fi
 fi
 
-fi
```

Comment:
...was there a "fi" to much?

Reply:
```text
if [ "$P1" != "$P2" ]; then
	echo "Passwords do not match"
	exit 1

if [ "$HASH" = "1" ]; then
	P1=$(printf %s "$P1" | sha256sum | awk '{print $1}') 
fi

fi
```

The above "if inside if" was problematic. Separating those two solved it. You can try to add simple echo functions to original code to see it.

Reply:
Ah ok. thx.

if you are doing ``exit 1`` you can do ``fi`` immediately. True.